account_bank_statement_import/account_bank_statement_import.py

```diff
@@ -89,90 +112,92 @@ def _import_file(self, data_file):
                 account_number, company_id=company_id,
                 currency_id=currency_id).id
         # Find or create the bank journal
-        journal_id = self._get_journal(
-            currency_id, bank_account_id, account_number)
+        journal_id = self._get_journal(currency_id, bank_account_id)
+        # By now journal and account_number must be known
+        if not journal_id:
+            raise Warning(_('Can not determine journal for import.'))
         # Prepare statement data to be used for bank statements creation
-        stmts_vals = self._complete_stmts_vals(
-            stmts_vals, journal_id, account_number)
-        # Create the bank statements
-        return self._create_bank_statements(stmts_vals)
+        statement = self._complete_statement(
```

Comment:
Why do you rename stmts_vals by statement. IMO it's confusing. With the new API with such a name, I expect to recieve a browse_record but in fact it's a dict.